code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm

```diff
@@ -128,6 +141,17 @@
 	recovery_time = world.time + buffer_time
 	ranged_cooldown = world.time + buffer_time
 
+/// This proc is called by the HRD-MDE grenade to enrage the megafauna. This should increase the megafaunas attack speed if possible, give it new moves, or disable weak moves. This should be reverseable, and reverses on zlvl change.
+/mob/living/simple_animal/hostile/megafauna/proc/enrage()
+	if(enraged || (health / maxHealth) * 100 <= 80 )
+		return
+	enraged = TRUE
+
+/mob/living/simple_animal/hostile/megafauna/proc/unrage()
+	if(!enraged)
+		return
+	enraged = FALSE
```

Comment:
```suggestion
	enraged = FALSE
```

Reply:
Parent is called on all unrages, would rather not unrage effects be ran if the fauna is not enraged

Reply:
That will still happen anyways, this return isnt doing anything. Returning in a parent proc wont stop the children proc from running

Reply:
Ah.